src/main/java/edu/kis/powp/jobs2d/command/DriverCommandCallCounterVisitor.java

```diff
@@ -1,20 +1,37 @@
 package edu.kis.powp.jobs2d.command;
 
-import java.util.Iterator;
 
+import java.util.Iterator;
 
 /**
- * Visitor class for driver commands.
+ * DriverCommandVisitor class for driver commands.
  */
-public class DriverCommandCallCounterVisitor implements Visitor {
+public class DriverCommandCallCounterVisitor implements DriverCommandVisitor {
 	private int counter = 0;
 
 	public int getCounter(){
 		return this.counter;
 	}
 
 	@Override
-	public void visit(DriverCommand driverCommand) {
-	    counter++;
+	public void visit(ICompoundCommand driverCommand) {
+		Iterator<DriverCommand> commandsIterator = driverCommand.iterator();
+		while (commandsIterator.hasNext()) {
+			DriverCommand partOfCompoundCommand = commandsIterator.next();
+			if (partOfCompoundCommand instanceof ICompoundCommand) {
+				this.visit((ICompoundCommand)partOfCompoundCommand);
+			}
+			this.counter++;

```

Comment:
this is actually a form of code duplication - finally should be removed for the visitor to work correctly